### PR TITLE
XAI: Add softmax activation for cls head output in DetClassProbabilityMapHook

### PR DESCRIPTION
**Pull request by negvet**
### Summary
Convert raw class scores to distribution with softmax activation.
See the improvement in the fig. below
image

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

**Review by sovrasov — APPROVED** [review]

**Review by jaegukhyun — APPROVED** [review]
Just a question, detection models often use multi-binary cross entropy loss for classification. Is there difference when we calculate cls_score through sigmoid for each category?

**Comment by sovrasov** [comment]
> Just a question, detection models often use multi-binary cross entropy loss for classification. Is there difference when we calculate cls_score through sigmoid for each category?

In this PR, softmax operation doesn't correspond to any loss function and performed only to normalize activations. @negvet correct me if I'm wrong

**Comment by negvet** [comment]
@jaegukhyun @sovrasov sorry for a late response, Vlad is correct, softmax is just for normalization of raw cls activations
